### PR TITLE
Add `quotedEnabled` feature flag

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Link
[needed for SwiftUI issue](https://stream-io.atlassian.net/browse/SWUI-126?atlOrigin=eyJpIjoiMWY1ODlhYmNiNWU0NDIzM2I5OGZjNjkyODMzMGEzYmEiLCJwIjoiaiJ9)

### 🎯 Goal

The `quotes` property was not taken in consideration in our `ChannelConfig`. This prevents the disabling of the quotes to be reflected in the iOS clients.

### 🛠 Implementation

Introduced a new property `quotesEnabled` in the `ChannelConfig`. Fixed the mock responses and the tests to include the `quote` property from the REST API.

### 📝 Changes

The most important change is the introduction of the `quotesEnabled` property.

### 🎨 UI Changes

No UI changes.

### 🧪 Testing

Set a breakpoint in `ChannelConfig`'s `init(from decoder: Decoder)` method. The new `quotesEnabled` property should be true by default. Go to the dashboard and try disabling/re-enabling the `Quotes` switch. The value should change accordingly.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎉 GIF

Alt Text

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # Upcoming
 ## StreamChat
+### ✅ Added
+- `quotesEnabled` property is added to the `ChannelConfig`.
```
Comment (evsaev) [inline CHANGELOG.md:7]:
```suggestion
- `quotesEnabled` property is added to the `ChannelConfig` [#1891](https://github.com/GetStream/stream-chat-swift/issues/1891)
```

**Review by evsaev — APPROVED** [review]
🚢

**Review by adamrushy — APPROVED** [review]
Looks good ✅

File: Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift [hunk 2]
```diff
@@ -81,9 +81,11 @@ open class ChatMessageActionsVC: _ViewController, ThemeProvider {
...
         switch message.localState {
         case nil:
-            var actions: [ChatMessageActionItem] = [
-                inlineReplyActionItem()
-            ]
+            var actions: [ChatMessageActionItem] = []
+            
+            if channelConfig.quotesEnabled {
+                actions.append(inlineReplyActionItem())
+            }
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift:88]: Can we add a test for this in `ChatMessageActionsVC_Tests.swift`? We have an example already for "mutesEnabled", should be a simple one 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # Upcoming
 ## StreamChat
+### ✅ Added
+- `quotesEnabled` property is added to the `ChannelConfig` [#1891](https://github.com/GetStream/stream-chat-swift/issues/1891)
```
Comment (nuno-vieira) [inline CHANGELOG.md:7]: Can we add an entry to ## StreamChatUI as well? 🙏

File: Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift (hunk 2 above)
Comment (martinmitrevski) [reply Sources/StreamChatUI/MessageActionsPopup/ChatMessageActionsVC.swift:88]: Yes, good point. Added now 👍

File: CHANGELOG.md (hunk 3 above)
Comment (martinmitrevski) [reply CHANGELOG.md:7]: Sure, updated the CHANGELOG.

**Review by nuno-vieira — APPROVED** [review]
LGTM! 🚀

**Pull request merged**